src/brsTypes/components/RoSGNode.ts

```diff
@@ -1914,3 +1925,55 @@ function addChildren(
         }
     });
 }
+
+function cloneNode(
+    interpreter: Interpreter,
+    toBeCloned: RoSGNode,
+    isDeepCopy: Boolean
+): RoSGNode | BrsInvalid {
+    let copy = createNodeByType(interpreter, new BrsString(toBeCloned.nodeSubtype));
+
+    let originalFields = toBeCloned.getFields();
+    let copiedFields = new RoAssociativeArray([]);
+    let addReplace = copiedFields.getMethod("addreplace");
+
+    // Copy the fields.
+    for (let [key, value] of originalFields) {
+        if (addReplace) {
+            addReplace.call(
+                interpreter,
+                new BrsString(key),
+                getBrsValueFromFieldType(value.getType(), value.toString())
+            );
+        }
+    }
+
+    if (!(copy instanceof BrsInvalid)) {
+        let update = copy.getMethod("update");
+
+        if (update) {
+            update.call(interpreter, copiedFields, BrsBoolean.True);
+        }
+
+        // A deep clone also copies children.
+        if (isDeepCopy) {
+            let appendChild = copy.getMethod("appendchild");
```

Comment:
I think we can use the `appendChildToParent` function directly to remove this line and replace the one below with:
```
copy.appendChildToParent(childCopy);
```

Reply:
@strattonbrazil I think I would have to expose the `appendChildtoParent` method to be able to use it here. Since I don't see a precedent for this in the `addChildren` function above which also uses the `appendChild` callable, I think I will leave it as is. Let me know if you feel strongly about this one though!